### DISCUSSION TITLE
better-sqlite3 is too aggressive on GLIBC-2.29 usage on linux and other platforms

### DISCUSSION THREAD

**Discussion by hazendaz** [renovatebot/renovate#27958] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Bitbucket

### Please tell us more about your question or problem

We are running RHEL 8.8 which has older GLIBC-2.28.  The only module in Renovate with problem that I've found is better-sqlite3.  I tried to go back to an older version but it gets dicey with nodejs usage (drop from 20 to 18 but does not fix because better-sqlite3 didn't fix correctly either - see https://github.com/WiseLibs/better-sqlite3/issues/403).  Since that item is still experimental for self hosted, would it be feasible to remove it until more users can catch up on platforms we are currently stuck on.  Maybe restore it in renovate 38?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Does anything prevent you running using our docker images?

**Reply by hazendaz** [reply]
Yes, our security restrictions are very rigid and docker is large on most images and mostly fail Synk vulnerability scans.  I can try again though and see if feasible.

Another option here might be to remove that to perform a temp release given renovate releases so fast then just return it right after.  That would unlock the blocker for us so we are up to date with something current here and we can revisit.  Unless I'm mistaken there isn't a lot of code involved here to do that.  Yet another option could be that I just fork it and remove that for us so we can continue on but that option is a lot more complicated.

**Reply by viceice** [reply]
@rarkins we can probably make sqlite optional and try to import at runtime only if required.

@hazendaz would that help you? awe won't remove that for one user, because we plan to use it by default for performance reasons.

**Reply by hazendaz** [reply]
As long as we can compile it ok it would yes.  I'm hoping this is a one time problem.  We are on the last 36 renovate now and want to get newer.  By the time we do this again we are hoping our Jeknins docker images are moved to RHEL 9.

**Reply by matthewwjones** [reply]
> @rarkins we can probably make sqlite optional and try to import at runtime only if required.

@viceice - that would be a big help for me too.  We're currently blocked from upgrading beyond the version that introduced sqlite because of platform specific binary restrictions.  We can't use a containerised version for similar reasons.

If sqlite was an optional dependency we would have access to upgrade to later versions.

**Reply by viceice** [reply]
- #28777

**Comment by viceice** [comment, marked as answer]
please try v37.334.0[^1]

[^1]: https://github.com/renovatebot/renovate/releases/tag/37.334.0

**Reply by matthewwjones** [reply]
@viceice I've tested the change you made in 37.334.0 - all looks good, that's a massive help - thank you!

**Reply by hazendaz** [reply]
Thanks @viceice, also confirmed resolved for us.
